### PR TITLE
Features/example sqllite inmemory

### PR DESCRIPTION
**Pull request by Rossall82**
Ben's database work. Rename sqlproj, add database tables for what we know so far.

### CODE REVIEW

**Comment by UKHOGreg** [comment]
Ah, merge commit bd280d2. Oh well!

File: src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Net.Mime;
+
+namespace Database.SQL.EF
+{
+    public class TasksDbBuilder : ICanCreateTables, ICanPopulateTables, ICanSaveChanges
```
Comment (benbhall) [inline src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs]: Now creating and seeing all tables so probably needs to be renamed now to reflect this.

Comment (benbhall) [reply src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs]: `WorkflowDbBuilder`?

File: src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using Database.SQL.EF.Models;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Data;
+using System.IO;
+using System.Net.Mime;
```
Comment (benbhall) [inline src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs]: System.Net.Mime; unused

File: src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs (hunk 1 above)
Comment (UKHOGreg) [reply src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs]: `WorkflowDbBuilder` sounds good to me

File: src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs (hunk 2 above)
Comment (UKHOGreg) [reply src/Databases/Database.SQL.EF/TasksTestingDbBuilder.cs]: We should remove that then

**Comment by benbhall** [comment]
> Is this to merge for a completed story?
> 
> Do you think we ought to check if it still works when the relationships are in place?  Will probably have to add FKs on an ALTER rather than CREATE as we're not controlling the order in which tables created.

I'm thinking that this is only complete when all the models are in place and registered in the DbContext. Ideally with more seeding and proof of relationships?

**Comment by SamirHassoun** [comment]
@benbhall I think these should be part of the next "Story 146824: Develop EF Core solution with local DB for smoke testing"

**Comment by benbhall** [comment]
> @benbhall I think these should be part of the next "Story 146824: Develop EF Core solution with local DB for smoke testing"

Ah right I thought you were already on that story. This is certainly quite far out of scope from the SQL Db + unit test project one. I think they've become one now :) Happy to merge this in then so that you have some separation.

**Comment by SamirHassoun** [comment]
> @benbhall I think these should be part of the next "Story 146824: Develop EF Core solution with local DB for smoke testing"



> > @benbhall I think these should be part of the next "Story 146824: Develop EF Core solution with local DB for smoke testing"
> 
> Ah right I thought you were already on that story. This is certainly quite far out of scope from the SQL Db + unit test project one. I think they've become one now :) Happy to merge this in then so that you have some separation.

yes, will do all the tidying up within the next story, thanks

**Pull request merged**